storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java

```diff
@@ -2214,45 +2207,62 @@ private void assertBlobSidecarsAvailabilityExceptPruned(
       final Collection<SignedBeaconBlock> availableBlocksSidecars,
       final Collection<SignedBeaconBlock> prunedBlocksSidecars) {
     availableBlocksSidecars.forEach(
-        block -> {
-          assertThat(database.getBlobSidecarKeys(block.getSlotAndBlockRoot())).isNotEmpty();
-        });
+        block -> assertThat(database.getBlobSidecarKeys(block.getSlotAndBlockRoot())).isNotEmpty());
     prunedBlocksSidecars.forEach(
-        block -> {
-          assertThat(database.getBlobSidecarKeys(block.getSlotAndBlockRoot())).isEmpty();
-        });
+        block -> assertThat(database.getBlobSidecarKeys(block.getSlotAndBlockRoot())).isEmpty());
   }
 
   private void addBlocks(final SignedBlockAndState... blocks) {
     addBlocks(Arrays.asList(blocks));
   }
 
   private void addBlocks(final List<SignedBlockAndState> blocks) {
+    addBlocksAndBlobSidecars(blocks, List.of());
+  }
+
+  private void addBlocksAndBlobSidecars(
+      final List<SignedBlockAndState> blocks, final List<BlobSidecar> blobSidecars) {
     final StoreTransaction transaction = recentChainData.startStoreTransaction();
     for (SignedBlockAndState block : blocks) {
-      transaction.putBlockAndState(block, spec.calculateBlockCheckpoints(block.getState()));
+      transaction.putBlockAndState(
+          block,
+          blobSidecars.stream()
+              .filter(blobSidecar -> blobSidecar.getBlockRoot().equals(block.getRoot()))
+              .collect(Collectors.toUnmodifiableList()),
+          spec.calculateBlockCheckpoints(block.getState()));
     }
     commit(transaction);
   }
 
-  private void addBlobSidecars(final List<BlobSidecar> blobSidecars) {
-    blobSidecars.forEach(blobSidecar -> database.storeBlobSidecar(blobSidecar));
+  private void add(final Collection<SignedBlockAndState> blocks) {
+    final StoreTransaction transaction = recentChainData.startStoreTransaction();
```

Comment:
Maybe one line `add(blocks, List.of());` instead?

Reply:
ah sorry, now I understand :)